### ISSUE TITLE
Linq parser error when soft deletes are configured and multiple where clauses are applied

### ISSUE THREAD

**Issue by jeffdoolittle** [JasperFx/marten#844]
This is a regression from 1.x to 2.x

This used to work in Marten 1.x:

```csharp
var result = await session
    .Query<MyDocument>()
    .Where(x => x.TenantId == request.TenantId)
    .Where(x => x.Id == request.Id)
    .SingleOrDefaultAsync();
```

But now only this works with 2.x:

```csharp
var result = await session
    .Query<MyDocument>()
    .Where(x => x.TenantId == request.TenantId && x.Id == request.Id)
    .SingleOrDefaultAsync();
```

The exception stack trace is:

```
System.NotSupportedException: Collection was of a fixed size.
   at System.SZArrayHelper.Add[T](T value)
   at Marten.Linq.WhereFragmentExtensions.Append(IWhereFragment fragment, IWhereFragment other)
   at Marten.Linq.WhereFragmentExtensions.Append(IWhereFragment fragment, IWhereFragment[] others)
   at Marten.Schema.DocumentMapping.FilterDocuments(QueryModel model, IWhereFragment query)
   at Marten.Linq.Model.LinqQuery`1.buildWhereFragment()
   at Marten.Linq.Model.LinqQuery`1..ctor(DocumentStore store, QueryModel model, IIncludeJoin[] joins, QueryStatistics stats)
   at Marten.Linq.MartenQueryable`1.executeAsync[TResult](Func`2 source, CancellationToken token)
   at Marten.Linq.MartenQueryable`1.ToListAsync[TResult](CancellationToken token)
   at Marten.QueryableExtensions.ToListAsync[T](IQueryable`1 queryable, CancellationToken token)
```

**Comment by jeffdoolittle** [comment]
Testing revealed that the issue arises when `SoftDelete` is configured for the Document Type.

**Comment by jeremydmiller** [comment]
Easy money, it's fixed and will be in 2.1 later today.
